### PR TITLE
Fix auto getProvince on sms-gateway meeting search

### PR DESCRIPTION
**Pull request by spiritualtheory**
Added check for comma in address to prevent adding local province/state code when user is searching for meetings out of state.

### CODE REVIEW

**Comment by radius314** [comment]
Same comment as before, 3.6.1 was released already.  Also it appears that this is showing a lot of change, something is off with your spacing.

**Comment by radius314** [comment]
The diff should look more like this https://github.com/bmlt-enabled/yap/pull/369/files?utf8=✓&diff=unified&w=1.

Also checking for the presence of a comma in general might be saner.

**Comment by spiritualtheory** [comment]
Started off with checking for just a comma, but thought that some may put in street addresses with cities - and even if that was an edge case, I didn't want to interfere with that functionality later. Figured that in most cases, a comma will appear in -3 or -4 from the end. 

On the spacing - I'm at work, so no IntelliJ here - just notepad++, notepad, and gitkraken. In Notepad++, I can see no spacing issues - it looks clearly like I just replaced

$coordinates = getCoordinatesForAddress($address . "," . getProvince());

with

if (str_exists(substr($address, -4), ',')) {
    $coordinates = getCoordinatesForAddress($address);
} else {
    $coordinates = getCoordinatesForAddress($address . "," . getProvince());
}

Original and proposed change look clean.

Opened both in notepad since that does no formatting whatsoever, the committed file shows standard lines/end of line whereas the master file I downloaded to compare to shows no line ends for some reason. Not sure why that would be.

**Comment by spiritualtheory** [comment]
The releasenotes update was based on request from Patrick - I added it under the last version number - I assumed that someone else is in charge of release / version numbers. Obviously no opposition to moving whatever wherever on this.

**Comment by radius314** [comment]
3.6.2 is the likely next version, add the change under that (as 3.6.1 was released on Jan 5, 2020).  Also bump the version in functions.php.  

I'm a bit confused by the logic of 3 or 4 from the end.  What if someone put a zip code at the end (after the state)?  What about if someone entered a country?

**Comment by spiritualtheory** [comment]
It can easily be set to just look for a comma in general - which was the original way I did it - but I was concerned with interfering with street address searches. If that's not a concern then it makes sense not to limit the search depth.

**Comment by spiritualtheory** [comment]
Edited as suggested

File: RELEASENOTES.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Release Notes
 
+### 3.6.2 (January 9, 2019)
+* Added simple check for state/province in SMS meeting search. [#369]
+
 ### 3.6.1 (January 5, 2019)
 * Fix for null coordinate responses in map data. [#363]
 * Fix for service body name in reports.
+* Added simple check for state/province in SMS meeting search. [#369]
```
Comment (radius314) [inline RELEASENOTES.md]: This need to be removed.

Comment (spiritualtheory) [reply RELEASENOTES.md]: Fixed

File: RELEASENOTES.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Release Notes
 
+### 3.6.2 (January 9, 2019)
```
Comment (radius314) [inline RELEASENOTES.md]: please remove the date and put `UNRELEASED`, the release date is put right before the release happens.

Comment (spiritualtheory) [reply RELEASENOTES.md]: Done

**Comment by spiritualtheory** [comment]
Positioning check removed, release date changed to UNRELEASED, functions updated to 3.6.2

**Comment by spiritualtheory** [comment]
sms-gateway changes clean now

**Comment by pjaudiomv** [comment]
may be a good candidate for a regex? something similar to this
http://refiddle.com/or36

**Comment by radius314** [comment]
I like the regex approach.

**Comment by spiritualtheory** [comment]
functionally the only difference this and the original check up to 4 positions back for comma version is this catches a space without comma - it does not answer the initial concerns that were presented as objections to the comma check being only at the last 4 positions - same issues apply here (what if address, zip; what if , country and not country code)

With that said, no objection to putting the suggested regex in to replace the comma check with the only difference being a-Z to include lowercase in that match.

**Comment by pjaudiomv** [comment]
The idea here is to not add the state of its not necessary, check the refer fiddle link above it will also catch if a zip is appended after a state (two letters) but yea it would t catch 123 Easy St,  01390. The regex probably should be adjusted to catch those too a a-z followed by comma space just comma or just sos e followed by 4 digits. (Not 5). It gets complicated  some countries have letters in their postal code and some use 4numbers. There's entire libraries dedicated to this

**Comment by pjaudiomv** [comment]
Maybe add an option to just turn off appending province. The best way to parse an address is by doing what we are doing, send it to google to cleanup. It's the appending of state that adds the monkey wrench. However as we know from the simpsons if you just send google Springfield with a country bias of us, who knows what state it will return  if there isn't a province bias

**Comment by spiritualtheory** [comment]
I think leaving the auto province helps more and theres fewer instances of needing the workaround - I ran into the issue because a homegroup member is getting ready to go to Columbus OH and wanted to look for what meetings he should expect on homegroup night - and our phone number returned meetings from NYC. I can see this being more of an issue for folks closer to state borders - but still, the majority of searches will just be local cities and will need the province add.

I like your regex and think it’s better than the initial check for comma within 4 of the end. My only point was the objections cited for limiting to the last 4 for a comma check all still apply to the regex because it has that same limitation - just catches spaces without commas which is helpful. 

Either way, I’m neutral - just looking for the minor bug to be fixed - and any of the 3 current options work - though actually a combination regex to check for space then 2 letter code with an or find comma anywhere might be a good compromise.

**Comment by spiritualtheory** [comment]
I’ll put the regex in in a bit with the catch for lowercase - I think that will catch the majority of uses and it can easily be amended should the need arise.

**Comment by spiritualtheory** [comment]
The pattern search looking for 2 characters following a space catches multi-word city names that have 2 letter words in the middle of their names. Not super common in the US though we have our in’s and on’s - but places like France and Brazil with du, de, do contained in many names would trigger this. I think a comma check stays the simplest way to interfere the least and still provide a stable workaround.

**Comment by radius314** [comment]
This sounds good.  Update the PR with the regex and I'll approve it.

**Comment by spiritualtheory** [comment]
No, last comment talked about how the regex would not work especially in other countries. It will flag many false positives where cities have du do in on de ... etc in the middle of their name because the regex pattern will catch letter, space, 2 letters. I think leaving it as a simple comma check as it is right now will yield the fewest false positives and best cover what we’re trying to do here. A comma at least indicates that the user is trying to be specific in some way.

**Comment by radius314** [comment]
Fair enough, either way this will have to be adjusted in the future.  Someone else will want it to behave slightly differently for their use-case.

**Comment by spiritualtheory** [comment]
Agreed

**Pull request merged**